### PR TITLE
Raise the Java requirement to 11 from 1.8 to be able to simplify code

### LINKED ISSUE #3322: source code build error
Hi,
I tried to build the latest source code.
But it failed with below error.
Can you check it?

```
Configuration on demand is an incubating feature.

> Configure project :
Reckoned version: 0.1.0-SNAPSHOT

> Configure project :reporter-web-app
Will use the Node executable file from '/home/jiyeongseok/.gradle/nodejs/node-v12.16.1-linux-x64/bin/node'.
Will use the Yarn JavaScript file from '/home/jiyeongseok/.gradle/yarn/yarn-v1.22.4/bin/yarn.js'.

> Task :downloader:compileKotlin FAILED
e: /home/jiyeongseok/ort/downloader/src/main/kotlin/vcs/Subversion.kt: (268, 46): Unresolved reference: requestPasswordAuthenticationInstance

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':downloader:compileKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleKotlinCompilerWork
   > Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 8s
32 actionable tasks: 6 executed, 26 up-to-date

```

### PR DESCRIPTION
**Pull request by sschuberth**
And to be able to use the requestPasswordAuthenticationInstance() method
from the Java Authenticator.

Fixes #3322.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -83,8 +83,8 @@ Change into the directory with ORT's source code and run `docker build -t ort .`
 
 Install these additional prerequisites:
 
-* OpenJDK 8 or Oracle JDK 8u161 or later (not the JRE as you need the `javac` compiler); also remember to set the
-  `JAVA_HOME` environment variable accordingly.
+* Java Development Kit (JDK) version 11 or later (not the JRE as you need the `javac` compiler); also remember to set
```
Comment (MarcelBochtler) [inline README.md:86]: For Java 11 no JRE standalone is available anymore. Therefore you could drop the text in the parentheses.
See: https://www.heise.de/download/product/java-runtime-environment-jre-627 and https://www.heise.de/download/product/java-runtime-environment-jre-627 (german)

**Review by MarcelBochtler — CHANGES_REQUESTED** [review]
As this enforces Java 11, we may want to suppress these warnings in a separate PR:
```
❯ ort --help
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by retrofit2.Platform (file:/home/bom1wa3/workspace/oss/ort/cli/build/install/ort/lib/retrofit-2.9.0.jar) to constructor java.lang.invoke.MethodHandles$Lookup(java.lang.Class,int)
WARNING: Please consider reporting this to the maintainers of retrofit2.Platform
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

Edit:
Looks like this cannot be simply disabled and it won't be a problem for users on Java 14+
https://github.com/square/retrofit/issues/3341

File: README.md (hunk 1 above)
Comment (sschuberth) [reply README.md]: Done.

File: README.md [hunk 2]
```diff
@@ -575,10 +574,9 @@ ORT is being continuously used on Linux, Windows and macOS by the
 [core development team](https://github.com/orgs/oss-review-toolkit/people), so these operating systems are
 considered to be well supported.
 
-To run the ORT binaries (also see [Installation from binaries](#from-binaries)) at least a Java Runtime Environment
-(JRE) version 8 is required, but using version 11 is recommended. Memory and CPU requirements vary depending on the size
-and type of project(s) to analyze / scan, but the general recommendation is to configure the JRE with 8 GiB of memory
-(`-Xmx=8g`) and to use a CPU with at least 4 cores.
+To run the ORT binaries (also see [Installation from binaries](#from-binaries)) at least a Java 11 is required. Memory
```
Comment (MarcelBochtler) [inline README.md:577]: `at least a Java 11 is required`

--> `at least Java 11 is required`

**Review by MarcelBochtler — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (sschuberth) [reply README.md]: Done.

**Review by MarcelBochtler — APPROVED** [review]

**Pull request merged**